
syntax = "proto3";

import "google/protobuf/empty.proto";

package sophos;

service Sophos {

// Search
rpc search (SearchRequest) returns (stream SearchReply) {}

// Update
rpc update (UpdateRequest) returns (google.protobuf.Empty) {}

}

message SearchRequest
{
    bytes search_token = 1;
    fixed32 add_count = 2;
}

message SearchReply
{
    uint64 result = 1;
}

message UpdateRequest
{
    bytes update_token = 1;
    uint64 index = 2;
}
